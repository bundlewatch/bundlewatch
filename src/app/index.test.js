import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'

const networkMock = new MockAdapter(axios)

import bundlewatchApi from '.'

describe(`bundlewatch Node API`, () => {
    it('Works with basic options', async () => {
        const result = await bundlewatchApi({
            files: [
                {
                    path: './__testdata__/*.jpg',
                    maxSize: '100kB',
                },
            ],
            defaultCompression: 'none',
        })

        // TODO: assert logger.warn called

        delete result.url
        expect(result).toMatchSnapshot()
    })

    it(`Works when files dont exist, shows warning`, async () => {
        const result = await bundlewatchApi({
            files: [
                {
                    path: './__testdata__/test-file-doesnt-exist.jpg',
                    maxSize: '100kB',
                },
            ],
        })

        delete result.url
        expect(result).toMatchSnapshot()
    })

    it('Works with CI environment', async () => {
        const MOCK_AUTH_TOKEN = 'mock-auth-token'
        const MOCK_REPO = {
            owner: 'mockowner',
            name: 'mockname',
            currentBranch: 'mockCurrentBranch',
            branchBase: 'mockBranchBase',
            commitSha: 'mockCommitsha',
        }

        networkMock
            .onPost('https://service.bundlewatch.io/store/lookup')
            .reply(200, {
                fileDetailsByPath: {
                    './__testdata__/test-file-1.jpg': {
                        size: 25000,
                        compression: 'gzip',
                    },
                    './__testdata__/test-file-deleted.jpg': {
                        size: 10000,
                        compression: 'gzip',
                    },
                },
            })

        // TODO: assert save was called

        const result = await bundlewatchApi({
            files: [
                {
                    path: './__testdata__/*.jpg',
                    maxSize: '1MB',
                },
            ],
            ci: {
                githubAccessToken: MOCK_AUTH_TOKEN,
                repoOwner: MOCK_REPO.owner,
                repoName: MOCK_REPO.name,
                repoCurrentBranch: MOCK_REPO.currentBranch,
                repoBranchBase: MOCK_REPO.branchBase,
                commitSha: MOCK_REPO.branchBase,
            },
        })

        delete result.url
        expect(result).toMatchSnapshot()
    })

    it('Throws validations error when using brotli compression without the package', async () => {
        let error
        try {
            await bundlewatchApi({
                files: [
                    {
                        path: './__testdata__/*.jpg',
                        maxSize: '100kB',
                    },
                ],
                defaultCompression: 'brotli',
            })
        } catch (e) {
            error = e
        }
        expect(error).toMatchSnapshot()
    })

    it('Normalizes hash when given a normalizeFilenames option', async () => {
        const result = await bundlewatchApi({
            files: [
                {
                    path: './__testdata__/*.js',
                    maxSize: '100kB',
                },
            ],
            defaultCompression: 'none',
            normalizeFilenames: '^.+?(\\.\\w+)\\.(?:js|css)$',
        })

        delete result.url
        expect(result.fullResults[0].filePath).toMatchInlineSnapshot(
            `"./__testdata__/test.bundle.js"`,
        )
    })
})
