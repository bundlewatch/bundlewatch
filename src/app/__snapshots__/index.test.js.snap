// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bundlewatch Node API Throws validations error when using brotli compression without the package 1`] = `
ValidationError {
  "message": "'brotli-size' package has not been installed, please install package to use this compression type",
  "type": "ValidationError",
}
`;

exports[`bundlewatch Node API Works when files dont exist, shows warning 1`] = `
Object {
  "fullResults": Array [
    Object {
      "error": "There is no matching file for ./__testdata__/test-file-doesnt-exist.jpg",
      "filePath": "./__testdata__/test-file-doesnt-exist.jpg",
      "status": "fail",
    },
  ],
  "status": "fail",
  "summary": "maxSize check failed",
}
`;

exports[`bundlewatch Node API Works with CI environment 1`] = `
Object {
  "fullResults": Array [
    Object {
      "baseBranchSize": 25000,
      "filePath": "./__testdata__/test-file-1.jpg",
      "maxSize": 1048576,
      "message": "38.08KB < 1MB (13.67KB larger than mockBranchBase) (gzip)",
      "size": 38995,
      "status": "pass",
    },
    Object {
      "baseBranchSize": 0,
      "filePath": "./__testdata__/test-file-2.jpg",
      "maxSize": 1048576,
      "message": "248.5KB < 1MB (gzip)",
      "size": 254466,
      "status": "pass",
    },
    Object {
      "baseBranchSize": 10000,
      "filePath": "./__testdata__/test-file-deleted.jpg",
      "maxSize": 0,
      "message": "./__testdata__/test-file-deleted.jpg: File removed (9.77KB smaller than mockBranchBase) (gzip)",
      "size": 0,
      "status": "removed",
    },
  ],
  "status": "pass",
  "summary": "Everything is in check (+262.17KB, -9.77KB, +738.5%)",
}
`;

exports[`bundlewatch Node API Works with basic options 1`] = `
Object {
  "fullResults": Array [
    Object {
      "baseBranchSize": 0,
      "filePath": "./__testdata__/test-file-1.jpg",
      "maxSize": 102400,
      "message": "38.05KB < 100KB (no compression)",
      "size": 38962,
      "status": "pass",
    },
    Object {
      "baseBranchSize": 0,
      "filePath": "./__testdata__/test-file-2.jpg",
      "maxSize": 102400,
      "message": "248.79KB > 100KB (no compression)",
      "size": 254765,
      "status": "fail",
    },
  ],
  "status": "fail",
  "summary": "maxSize check failed",
}
`;
